### PR TITLE
Configure --enable-pdns-option --with-third-party-module

### PR DESCRIPTION
**Pull request by jsoref**
### Short description
https://github.com/PowerDNS/pdns/wiki/Yak-list
> <lieter> another yak I have with configure is our inconsistent use of --enable-X and --with-X
> <lieter> --enable are _our_ programs's options and --with-X should be external deps

### Checklist
I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: docs/appendices/compiling.rst [hunk 1]
```diff
@@ -62,13 +62,13 @@ ed25519 support with libsodium
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 The PowerDNS Authoritative Server can link with `libsodium <https://download.libsodium.org/doc/>`_ to support ed25519 (DNSSEC algorithm 15).
-To detect libsodium, use the ``--enable-libsodium`` configure option.
```
Comment (pieterlexis) [inline docs/appendices/compiling.rst]: perhaps mention this (also for the other doc changes):

```rst
.. versionchanged:: 4.2.0
  This option was called ``--enable-libsobium`` before
```

**Comment by rgacogne** [comment]
PR is conflicted though :-/

**Comment by jsoref** [comment]
@rgacogne: with some additional effort, we could probably support the old flag names... I don't remember the syntax.

**Comment by rubenk** [comment]
I think this is long overdue, thanks. As a downstream packager I don't mind changing some things in spec files.

**Pull request merged**